pandas/core/dtypes/cast.py

```diff
@@ -1910,6 +1910,9 @@ def validate_numeric_casting(dtype: np.dtype, value: Scalar) -> None:
             raise ValueError(
                 f"Cannot assign {type(value).__name__} to float/integer series"
             )
+    if dtype.kind == "b":
+        if is_scalar(value) and not is_bool(value):
```

Comment:
worth making these not compound? both L1913-1914 and L1907-1908 ?

Reply:
i dont think so; i expect were going to have to flesh out the dtype.kind == "b" case for non-scalars before too long (hopefully can dedup with can_hold_element)